pkgs/apps/bagel/default.nix

```diff
@@ -44,7 +43,6 @@ in stdenv.mkDerivation rec {
     "-O3"
     "-DCOMPILE_J_ORB"
   ] ++ lib.lists.optionals (!useMKL) [ "-lblas" "-llapack" ]
-    ++ lib.lists.optional optAVX "-mavx -mavx2"
```

Comment:
Is this supposed to be handled by the `march` optimisation from the `stdenv`? Wouldn't it be better to keep using the `stdenv.hostPlatform.avx2Support` flags directly? The impurity of `march` and `mtune` doesn't feel so nice.

Reply:
Yes, the idea is that the march/mtune flags are set in the new stdenv. This way more of the code is compiled with optimizations and we need to mention it in less places in derivations (unless of course it is a configure or cmake flag).
Note that the march/mtune flags are perfectly pure here, when set to a specific architecture. These optimizations only become impure when set to `-march=native`/`-mtune=native`.

Reply:
Ah I see, this makes a lot of sense. Thanks for the explanation.